embulk-api/src/main/java/org/embulk/spi/PageBuilder.java

```diff
@@ -127,19 +128,59 @@ public void setString(int columnIndex, String value) {
     }
 
     /**
+     * Sets a JSON value at the specified column in the {@code msgpack-java} representation.
+     *
+     * @param column  the column to set the JSON value
+     * @param value  the JSON value in the {@code msgpack-java} representation
+     * @deprecated Use {@link #setJson(org.embulk.spi.Column, org.embulk.spi.json.JsonValue)} instead.
+     *
      * @since 0.8.0
      */
+    @Deprecated
+    @SuppressWarnings("deprecation")
     public void setJson(Column column, Value value) {
         this.delegate.setJson(column, value);
     }
 
     /**
+     * Sets a JSON value at the specified column.
+     *
+     * @param column  the column to set the JSON value
+     * @param value  the JSON value
+     *
+     * @since 0.10.42
+     */
+    public void setJson(final Column column, final JsonValue value) {
+        this.delegate.setJson(column, value);
+    }
+
+    /**
+     * Sets a JSON value at the specified column in the {@code msgpack-java} representation.
```

Comment:
column -> column index?

Reply:
I don't think "column index" is adequate here. It "sets a JSON value at the specified column **specified by the index**", but the target is still a "column".